source/ShopPanel.cpp

```diff
@@ -660,6 +660,10 @@ bool ShopPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 		return DoScroll(Screen::Bottom());
 	else if(key == SDLK_PAGEDOWN)
 		return DoScroll(Screen::Top());
+	else if(key == SDLK_HOME)
+		return DoScroll(max(maxInfobarScroll, maxSidebarScroll));
```

Comment:
This doesn't work because you scroll value is different depending on the currently active panel. So I'd suggest creating two new functions, let's say `DoScrollTop` and `DoScrollBottom` that set the correct scroll value depending on the active panel (see `DoScroll`, it should be straightforward from there).